File: src-tauri/src/commands/llm.rs

#[tauri::command]
pub fn llm_check_availability() -> Result<serde_json::Value, String> {
    Ok(serde_json::json!({
        "success": true,
        "available": false,
        "hasCli": false,
        "binaryDownloaded": false,
        "models": [],
        "error": "LLM not yet implemented in Tauri backend"
    }))
}

#[tauri::command]
pub fn llm_get_models() -> Result<serde_json::Value, String> {
    Ok(serde_json::json!({
        "success": true,
        "models": [],
    }))
}

#[tauri::command]
pub fn llm_download_model(_model_name: String) -> Result<serde_json::Value, String> {
    Ok(serde_json::json!({
        "success": false,
        "error": "LLM download not yet implemented"
    }))
}

#[tauri::command]
pub fn llm_delete_model(_model_name: String) -> Result<serde_json::Value, String> {
    Ok(serde_json::json!({
        "success": false,
        "error": "LLM delete not yet implemented"
    }))
}

#[tauri::command]
pub fn llm_test_process(
    _text: String,
    _model_name: Option<String>,
) -> Result<serde_json::Value, String> {
    Ok(serde_json::json!({
        "success": false,
        "error": "LLM test not yet implemented"
    }))
}

#[tauri::command]
pub fn llm_get_models_path() -> Result<String, String> {
    Ok(String::new())
}

File: electron/preload/llm.ts
/**
 * LLM post-processing preload API.
 */
import { ipcRenderer } from 'electron';
import type { ElectronAPISection } from './types';

export function createLlmAPI(): ElectronAPISection {
  return {
    llmCheckAvailability: () => ipcRenderer.invoke('llm-check-availability'),
    llmGetModels: () => ipcRenderer.invoke('llm-get-models'),
    llmDownloadModel: (modelName) => ipcRenderer.invoke('llm-download-model', modelName),
    llmDeleteModel: (modelName) => ipcRenderer.invoke('llm-delete-model', modelName),
    llmPauseDownload: () => ipcRenderer.invoke('llm-pause-download'),
    llmResumeDownload: () => ipcRenderer.invoke('llm-resume-download'),
    llmCancelDownload: () => ipcRenderer.invoke('llm-cancel-download'),
    llmGetDownloadState: () => ipcRenderer.invoke('llm-get-download-state'),
    llmTestProcess: (text, modelName) => ipcRenderer.invoke('llm-test-process', text, modelName),
    llmGetModelsPath: () => ipcRenderer.invoke('llm-get-models-path'),
    llmChooseModelsFolder: () => ipcRenderer.invoke('llm-choose-models-folder'),
    llmScanModelsFolder: () => ipcRenderer.invoke('llm-scan-models-folder'),
    llmDownloadBinary: () => ipcRenderer.invoke('llm-download-binary'),
    llmCancelBinaryDownload: () => ipcRenderer.invoke('llm-cancel-binary-download'),
    llmGetBinaryDownloadState: () => ipcRenderer.invoke('llm-get-binary-download-state'),
    llmCheckBinary: () => ipcRenderer.invoke('llm-check-binary'),
  };
}
